Sources/SourceKit/SourceKitServer.swift

```diff
@@ -523,3 +570,54 @@ public func languageService(
 
 public typealias Notification = LanguageServerProtocol.Notification
 public typealias Diagnostic = LanguageServerProtocol.Diagnostic
+
+extension IndexSymbolKind {
+  func toLspSymbolKind() -> SymbolKind {
+    switch self {
+    case Self.class:
+      return .class
+    case Self.classMethod,
+      Self.instanceMethod,
+      Self.staticMethod:
+      return .method
+    case Self.instanceProperty,
+      Self.staticProperty,
+      Self.classProperty:
+      return .property
+    case Self.enum:
+      return .enum
+    case Self.enumConstant: // Unsure about this one
+      return .enumMember
+    case Self.protocol:
+      return .interface
+    case Self.function,
+      Self.conversionFunction: // assuming conversion function is function
+      return .function
+    case Self.variable:
+      return .variable
+    case Self.struct:
+      return .struct
+    case Self.parameter:
+      return .typeParameter
+    case Self.parameter:
+      return .typeParameter
+
+    default:
+      return .null
+    }
+  }
+}
+
+extension SymbolOccurrence {
+  /// Get the name of the symbol that is a parent of this symbol, if one exists
+  func getContainerName() -> String? {
+    var foundChildRelation: SymbolRelation?
+    self.forEachRelation { relation in
+      if relation.roles.contains(.childOf) {
+        foundChildRelation = relation
+      }
+      return false
```

Comment:
Shouldn't this be `return false` if we found one, but `return true` to keep looking?